### PR TITLE
added element change to MCS

### PR DESCRIPTION
**Pull request by richardjgowers**

### CODE REVIEW

**Review by dwhswenson — COMMENTED** [review]
A couple issues related to SemVer.

Also, would there be any reason to allow users to also pass another rdkit AtomCompare, rather than just limiting it to these two options?

File: lomap/mcs.py [hunk 1]
```diff
@@ -90,6 +91,11 @@ def __init__(self, moli, molj, time=20, verbose='info', max3d=1000, threed=False
             ???, default 1,000
         threed : bool, optional
             ???, default False
+        element_change : bool, optional
+            whether to allow elemental changes in mappings, default True
+
+        ..versionchanged:: 2.0.0
+           Added element_change kwarg
```
[the remark this answers is not in the document]
Comment (dwhswenson) [reply lomap/mcs.py:98]: Wait, nvm.. I misread which was the old default

**Comment by richardjgowers** [comment]
Yeah we could allow an arbitrary `AtomCompare`, but then I'd need to think of representative tests for all the members of the enum

**Review by IAlibay — CHANGES_REQUESTED** [review]
Couple of small things - mostly some docs things

I understand that CompareAny will just take any atoms in the comparison, I assume CompareElements will restrict this to only alike elements, but I can't find the actual docs for it.

File: lomap/mcs.py [hunk 2]
```diff
@@ -580,9 +586,14 @@ def set_ring_counter(mol):
         # MCS calculation. In RDKit the MCS is a smart string. Ring atoms are
         # always mapped in ring atoms.
         # Don't add the mcs result as a member variable as it can't be pickled
+        if element_change:
+            atom_compare = rdFMCS.AtomCompare.CompareAny
+        else:
+            atom_compare = rdFMCS.AtomCompare.CompareElements
```
[the remark this answers is not in the document]
Comment (dwhswenson) [reply lomap/mcs.py:592]: I believe this is the actual cocde doing the work: https://github.com/rdkit/rdkit/blob/94cde75973ec423d44996f0d47352b4fe4e3a90a/Code/GraphMol/FMCS/FMCS.cpp#L247-L330

[the remark this answers is not in the document]
Comment (IAlibay) [reply lomap/mcs.py:592]: Awesome thanks, that makes sense now :)

File: lomap/dbmol.py [hunk 3]
```diff
@@ -93,7 +93,8 @@ class DBMolecules(object):
 
     # Initialization function
     def __init__(self, directory, parallel=1, verbose='off',
-                 time=20, ecrscore=0.0, threed=False, max3d=1000.0, output=False,
+                 time=20, ecrscore=0.0, threed=False, max3d=1000.0,
```
Comment (IAlibay) [inline lomap/dbmol.py:96]: Given CI didn't fail, I assume this isn't covered by tests at all?

File: lomap/mcs.py [hunk 4]
```diff
@@ -86,16 +88,24 @@ def __init__(self, moli, molj, time=20, verbose='info', max3d=1000, threed=False
             timeout on MCS, default 20
         verbose : str, optional
             logging level, default 'info'
-        max3d : int, optional
-            ???, default 1,000
+        max3d : float, optional
+            The MCS is trimmed to remove atoms which are further apart than
+            this distance, default 1,000.0
```
Comment (IAlibay) [inline lomap/mcs.py:93]:
```suggestion
            this distance (in units of Angstrom), default 1,000.0
```

**Review by IAlibay — APPROVED** [review]
Just the one thing, I'll approve now rather than block.

**Pull request merged**